src/plugins/discover/public/application/main/utils/get_data_view_by_text_based_query_lang.ts

```diff
@@ -33,6 +33,7 @@ export async function getDataViewByTextBasedQueryLang(
     currentDataView?.isPersisted() ||
     indexPatternFromQuery !== currentDataView?.getIndexPattern()
   ) {
+    // I could use DataViewLazy here BUT this function returns a DataView - where is that DataView being used?
```

Comment:
This is identified in https://github.com/elastic/kibana/issues/169835 and is where my changes start but they seem to ripple outward. Someone with more experience in the discover state management could probably help me come up with a strategy.

Reply:
I had a look at this, and we can't currently use DataViewLazy here, because this is setting the dataView in state which expects the DataView that's not so Lazy 

So there's one thing in ES|QL that needs to be resolved differently. When the query is changed, the change of index pattern /  data view is checked, and this can lead to loading time without any loading indication ... so we currently can't reduce field caps for ES|QL, but we could improve loading communication to the user, which happens here, when a query is submitted: 

https://github.com/elastic/kibana/blob/edd384a141dff680c17f2d4346e65a971b9fef16/src/plugins/discover/public/application/main/services/discover_data_state_container.ts#L299-L318